x-pack/solutions/security/plugins/elastic_assistant/server/lib/attack_discovery/schedules/types.ts

```diff
@@ -8,14 +8,33 @@
 import { RuleExecutorOptions, RuleType, RuleTypeState } from '@kbn/alerting-plugin/server';
 import { SecurityAttackDiscoveryAlert } from '@kbn/alerts-as-data-utils';
 import { AttackDiscoveryScheduleParams } from '@kbn/elastic-assistant-common';
+import { ALERT_ATTACK_DISCOVERY_USERS } from './fields';
+
+/**
+ * This is a WORKAROUND until the `createSchemaFromFieldMap` can handle complex mappings.
+ * Right now that tool cannot properly handle the combination of optional and required fields within the same nested/object field type.
+ * Instead of creating an intersection type it generates a separate fields in required and optional sections as separate flattened fields.
+ * As a workaround, we strip out incorrectly generated fields and re-add them in a correct format.
+ */
+export type AttackDiscoveryAlert = Omit<
```

Comment:
Consider naming this type `AttackDiscoveryAlertDocument` to avoid naming collisions with API type names

Reply:
Done in https://github.com/elastic/kibana/pull/218025/commits/5fac2e260e64a12d448a387ca411d598f2bccc58